intro-HPC/ch_MATLAB.tex

```diff
@@ -0,0 +1,173 @@
+\chapter{MATLAB}
+\label{ch:matlab}
+
+\section{Why is the MATLAB compiler required?}
+
+The main reason behind this alternative way of using MATLAB is licensing: only
+a limited number of MATLAB sessions can be active at the same time. However, once
+the MATLAB program is compiled using the MATLAB compiler, the resulting stand-alone
+executable can be run without needing to contact the license server.
+
+Note that a license is required for the MATLAB Compiler,
+see \url{https://nl.mathworks.com/help/compiler/index.html}. If the \lstinline|mcc|
+command is provided by the MATLAB installation you are using, the MATLAB compiler
+can be used as explained below.
+
+\ifgent
+Only a limited amount of MATLAB sessions can be active at the same time because
+there are only a limited amount of MATLAB research licences available on
+the \university MATLAB license server. If each job would need a license,
+licenses would quickly run out.
+\fi
+
+\section{How to compile MATLAB code}
+
+Compiling MATLAB code can only be done from the login nodes, because only login
+nodes can access the MATLAB license server, workernodes on clusters can not.
+
+To access the MATLAB compiler, the \lstinline|MATLAB| module should be loaded first. Make sure
+you are using the same \lstinline|MATLAB| version to compile and to run the compiled MATLAB
+program.
+
+\begin{prompt}
+%\shellcmd{module avail MATLAB}%
+----------------------%\modulelocation{}%----------------------
+   MATLAB/2016b    MATLAB/2017b    MATLAB/2018a (D)
+%\shellcmd{module load MATLAB/2018a}
+\end{prompt}
+
+After loading the \lstinline|MATLAB| module, the \lstinline|mcc| command can be used. To get help on
+\lstinline|mcc|, you can run \lstinline|mcc -?|.
+
+To compile a standalone application, the \lstinline|-m|
+flag is used (the \lstinline|-v| flag means verbose ouput).
+To show how \lstinline|mcc| can be used, we use the \lstinline|magicsquare| example
+that comes with MATLAB.
+
+First, we copy the \lstinline|magicsquare.m| example that comes with MATLAB to \lstinline|example.m|.
+
+
+\begin{prompt}
+%\shellcmd{cp \$EBROOTMATLAB/extern/examples/compiler/magicsquare.m example.m}%
```

Comment:
I would break this up, the clarify distinguish between copying the example and the actual compilation:

```
First, we copy the \lstinline|magicsquare.m| example that comes with MATLAB to \lstinline|example.m|:

\begin{prompt}
%\shellcmd{cp \$EBROOTMATLAB/extern/examples/compiler/magicsquare.m example.m}%
\end{prompt}

To compile a MATLAB program, use \lstinline|mcc -mv|:

\begin{prompt}
%\shellcmd{mcc -mv example.m}%
...
\end{prompt}

```